File: Programmers/Level 1/25.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    vector<int> student1 {1,2,3,4,5};
    vector<int> student2 {2,1,2,3,2,4,2,5};
    vector<int> student3 {3,3,1,1,2,2,4,4,5,5};
    
    int cnt1 = 0;
    int cnt2 = 0;
    int cnt3 = 0;
    
    for(int i = 0; i < answers.size(); i++) {
        int j = i % 5;
        int k = i % 8;
        int l = i % 10;
        
        if(answers[i] == student1[j]) cnt1++;
        if(answers[i] == student2[k]) cnt2++;
        if(answers[i] == student3[l]) cnt3++;
    }
    
    int ans = max(cnt1, max(cnt2, cnt3));
    
    if(ans == cnt1) answer.push_back(1);
    if(ans == cnt2) answer.push_back(2);
    if(ans == cnt3) answer.push_back(3);
    
    return answer;
}

File: Programmers/Level 1/25.py
def solution(answers):
    answer = []
    
    one = [1, 2, 3, 4, 5]
    two = [2, 1, 2, 3, 2, 4, 2, 5]
    three = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    s1 = 0 # 수포자1
    s2 = 0 # 수포자2
    s3 = 0 # 수포자3
    for i in range(len(answers)):
        if one[i%5] == answers[i]:
            s1 += 1
        if two[i%8] == answers[i]:
            s2 += 1
        if three[i%10] == answers[i]:
            s3 += 1
    total = max(s1, max(s2, s3))
    if total == s1:
        answer.append(1)
    if total == s2:
        answer.append(2)
    if total == s3:
        answer.append(3)
    return answer
